### PR TITLE
Real deleting accounts

### PR DESCRIPTION
**Pull request by gavofyork**
- [x] RPC
- [x] Test for RPC
- [x] parity.js binding
- [x] UI integration

### CODE REVIEW

**Comment by gavofyork** [comment]
Last two should be down to @jacogr

**Comment by jacogr** [comment]
Linked issue https://github.com/ethcore/parity/issues/3105

**Comment by gavofyork** [comment]
over to you, @jacogr .

**Comment by jacogr** [comment]
UI integration - https://youtu.be/4Ih7ITwbq-A

@gavofyork It would be good if we could have the same for the `address_book.json` interface. (Currently those are all marked as deleted, but not currently actually deleted)

Closes https://github.com/ethcore/parity/issues/3105

**Comment by gavofyork** [comment]
will do the address book deletion in a separate PR.

File: rpc/src/v1/impls/parity_accounts.rs [hunk 1]
```diff
@@ -117,6 +117,15 @@ impl<C: 'static> ParityAccounts for ParityAccountsClient<C> where C: MiningBlock
 			.map_err(|e| errors::account("Could not fetch account info.", e))
 	}
 
+	fn kill_account(&self, account: RpcH160, password: String) -> Result<bool, Error> {
+		try!(self.active());
+		let account: Address = account.into();
+		take_weak!(self.accounts)
+			.kill_account(&account, &password)
+			.map(|_| true)
+			.map_err(|e| errors::account("Could not fetch account info.", e))
```
Comment (tomusdrw) [inline rpc/src/v1/impls/parity_accounts.rs]: "Could not kill account?"

File: rpc/src/v1/tests/mocked/parity_accounts.rs [hunk 2]
```diff
@@ -116,3 +116,27 @@ fn should_be_able_to_set_meta() {
...
+	let response = r#"{"jsonrpc":"2.0","error":{"code":-32602,"message":"Invalid params","data":null},"id":1}"#;
+	let res = tester.io.handle_request_sync(&request);
+	assert_eq!(res, Some(response.into()));
+
+	let request = format!(r#"{{"jsonrpc": "2.0", "method": "parity_killAccount", "params": ["0x{}", "password"], "id": 1}}"#, address.hex());
+	let response = r#"{"jsonrpc":"2.0","result":true,"id":1}"#;
+	let res = tester.io.handle_request_sync(&request);
+	assert_eq!(res, Some(response.into()));
+
+	let request = r#"{"jsonrpc": "2.0", "method": "parity_accountsInfo", "params": [], "id": 1}"#;
```
Comment (tomusdrw) [inline rpc/src/v1/tests/mocked/parity_accounts.rs]: Could be just a check in `tester.accounts`, no?

File: js/src/modals/DeleteAccount/deleteAccount.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+    return (
+      <ConfirmDialog
+        className={ styles.body }
+        title='confirm removal'
+        visible
+        onDeny={ this.closeDeleteDialog }
+        onConfirm={ this.onDeleteConfirmed }>
+        <div className={ styles.hero }>
+          Are you sure you want to remove permanently delete the following account?
```
Comment (tomusdrw) [inline js/src/modals/DeleteAccount/deleteAccount.js]: "remove permanently delete"

File: js/src/modals/DeleteAccount/deleteAccount.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    api.parity
+      .killAccount(account.address, password)
+      .then((result) => {
+        if (result === true) {
+          router.push('/accounts');
+        } else {
+          newError(new Error('Deletion failed.'));
+        }
+
+        this.closeDeleteDialog();
```
Comment (tomusdrw) [inline js/src/modals/DeleteAccount/deleteAccount.js]: Do we really want to close the dialog if there is an error?

File: js/src/modals/DeleteAccount/deleteAccount.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+      })
+      .catch((error) => {
+        console.error('onDeleteConfirmed', error);
+        newError(new Error(`Deletion failed: ${error.message}`));
+        this.closeDeleteDialog();
```
Comment (tomusdrw) [inline js/src/modals/DeleteAccount/deleteAccount.js]: Do we really want to close the dialog if there is an error?

File: js/src/modals/DeleteAccount/deleteAccount.js [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+class DeleteAccount extends Component {
+  static contextTypes = {
+    api: PropTypes.object.isRequired,
+    router: PropTypes.object
+  }
+
+  static propTypes = {
+    account: PropTypes.object.isRequired,
+    onClose: PropTypes.func,
+    newError: PropTypes.func
```
Comment (tomusdrw) [inline js/src/modals/DeleteAccount/deleteAccount.js]: Why the functions are not required?

Consider  `static defaultProps = { onClose: () => {}, newError: () => () }`

File: rpc/src/v1/tests/mocked/parity_accounts.rs (hunk 2 above)
Comment (gavofyork) [inline rpc/src/v1/tests/mocked/parity_accounts.rs]: where is `tester.accounts`?

**Comment by gavofyork** [comment]
grumbles addressed on my part

**Pull request merged**